src/cider/tests/nextgen/benchmark/JITlibBenchmark.cpp

```diff
@@ -0,0 +1,338 @@
+
+template <JITTypeTag JITType, typename ExecKernel>
+void basicArithmeticExecFunc(benchmark::State& state, ExecKernel&& kernel) {
+  using NativeT = typename JITTypeTraits<JITType>::NativeType;
+
+  std::vector<NativeT> a_col(state.range(0), 1);
+  std::vector<NativeT> b_col(state.range(0), 1);
+  std::vector<NativeT> out_col(state.range(0), 1);
+
+  for (auto _ : state) {
+    benchmark::DoNotOptimize(
+        kernel(a_col.data(), b_col.data(), out_col.data(), state.range(0)));
+  }
+}
+
+template <JITTypeTag JITType>
+void jitlibBasicArithmeticExecFunc(benchmark::State& state, JITFunctionPointer& func) {
+  using NativeT = typename JITTypeTraits<JITType>::NativeType;
+  auto func_ptr = func->template getFunctionPointer<int,
+                                                    NativeT*,
+                                                    NativeT*,
+                                                    NativeT*,
+                                                    int64_t>();  // a_col, b_col, out, len
+
+  std::vector<NativeT> a_col(state.range(0), 1);
+  std::vector<NativeT> b_col(state.range(0), 1);
+  std::vector<NativeT> out_col(state.range(0), 1);
+
+  for (auto _ : state) {
+    benchmark::DoNotOptimize(
+        func_ptr(a_col.data(), b_col.data(), out_col.data(), state.range(0)));
+  }
+}
+
+#define CREATE_CPP_BASIC_ARITHMETRIC_BENCHMARK(JITTYPE, OP, OPNAME) \
+  void basic_arith_##OPNAME##_cpp(benchmark::State& state) {        \
+    basicArithmeticExecFunc<JITTYPE>(                               \
+        state,                                                      \
+        [](auto a, auto b, auto out, int64_t len)                   \
+            NEVER_INLINE DISABLE_AUTO_VECTORIZATION -> auto {       \
+              for (int64_t i = 0; i < len; ++i) {                   \
+                out[i] = a[i] OP b[i];                              \
+              }                                                     \
+              return out[len - 1];                                  \
+            });                                                     \
+  }                                                                 \
+  BENCHMARK(basic_arith_##OPNAME##_cpp)                             \
+      ->RangeMultiplier(1 << 10)                                    \
+      ->Range(1 << 10, 1 << 10);
+
+#define CREATE_JIT_BASIC_ARITHMETRIC_BENCHMARK(JITTYPE, OP, OPNAME)                      \
+  void basic_arith_##OPNAME##_jit(benchmark::State& state) {                             \
+    LLVMJITModule module(                                                                \
+        std::string("basic_arith_") + #OPNAME + getJITTypeName(JITTYPE) + "jit",         \
+        false,                                                                           \
+        CompilationOptions{.aggresive_jit_compile = true,                                \
+                           .dump_ir = true,                                              \
```

Comment:
close dump_ir for benchmark

Reply:
It's fine, IR build and JIT compile procedure is out of the benchmark loop.